### ISSUE TITLE
signIn() and signUp() does not trigger SupabaseAuthRequiredState nor SupabaseAuthState 

### ISSUE THREAD

**Issue by Shiba-Kar** [supabase-community/supabase-flutter#120]
SupaAuthState class

```dart
class SupaAuthState<T extends StatefulWidget> extends SupabaseAuthState<T> {
  @override
  void onUnauthenticated() {
    log("UnAuthorized");
    Beamer.of(context).beamToReplacementNamed('/login');
  }

  @override
  void onAuthenticated(Session session) {
    log("Authorized");
    Beamer.of(context).beamToReplacementNamed('/home');
  }

  @override
  void onPasswordRecovery(Session session) {
    log("Password Recovery");
  }

  @override
  void onErrorAuthenticating(String message) {
    //Scaffold.of(context).showBodyScrim(true, 10.0);
  }
}
```
Signin does not trigger onAuthenticated() though Session prints over the console . Showing that login worked.
On logingout the onUnauthenticated() works as expected !!
```dart

class LoginView extends StatefulWidget {
  const LoginView({Key? key}) : super(key: key);

  @override
  State<LoginView> createState() => _LoginViewState();
}

class _LoginViewState extends SupaAuthState<LoginView> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: [
            ElevatedButton(
              onPressed: () async {
                startAuthObserver();
                var res = await Supabase.instance.client.auth.signIn(
                  email: "shibaprasadkar50@gmail.com",
                  password: "password",
                );
                log(res.user?.email ?? "");
              },
              child: const Text('Login'),
            ),
            ElevatedButton(
              onPressed: () async {
                await Supabase.instance.client.auth.signOut();
              },
              child: const Text('Logout'),
            ),
          ],
        ),
      ),
    );
  }
}

```

**Comment by davidhoang05022009** [comment]
Having the same issue on Flutter 3.0.3, email and password auth on Linux, but restart the app will trigger onAuthenticated() (kinda weird).

**Comment by nextdev1111** [comment]
Same issue here

**Comment by davidhoang05022009** [comment]
I think this issud might be fixed via PR [#124](https://github.com/supabase-community/supabase-flutter/pull/124)
Just sit back and wait for the PR to be merge :)

**Comment by dshukertjr** [comment]
Just to post an update here, we are hoping to release #124 some time this week!
